### PR TITLE
Removed dev_timeout from junos_cli

### LINKED ISSUE #63599: [BUG] Deltaproxy and napalm.junos_cli
**Description**
We've encountered some issues with using the junos execution module with the junos napalm proxy. 

There seem to be some kind of autovivification-issue, or similar, affecting the timeout-wrapper in the junos execution module. The wrapper expects the kwarg "dev_timeout" to exist and be defined, when in reality it is 'None' when not specifying it.

```
salt hostname napalm.junos_cli 'show version' format=xml
hostname:
    Passed invalid arguments to napalm.junos_cli: '>' not supported between instances of 'int' and 'NoneType'
```

```
salt hostname napalm.junos_cli 'show version' format=xml dev_timeout=30
hostname:
    <works>
```

**Setup**
(Please provide relevant configs and/or SLS files (be sure to remove sensitive info. There is no general set-up of Salt.)

Please be as specific as possible and give set-up details.

- [ ] on-prem machine
- [ ] VM (Virtualbox, KVM, etc. please specify)
- [ ] VM running on a cloud service, please be explicit and add details
- [x] container (Kubernetes, Docker, containerd, etc. please specify): 1x Master container, 1x Deltaproxy container.
- [ ] or a combination, please be explicit
- [ ] jails if it is FreeBSD
- [ ] classic packaging
- [ ] onedir packaging
- [ ] used bootstrap to install

**Steps to Reproduce the behavior**
One master running in docker container.
One delta-proxy running in docker container.
One Junos MX204

Traceback from the host running the delta proxy:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/salt/metaproxy/deltaproxy.py", line 669, in thread_return
    opts, data, func, args, kwargs
  File "/usr/local/lib/python3.7/site-packages/salt/loader/lazy.py", line 149, in __call__
    return self.loader.run(run_func, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/salt/loader/lazy.py", line 1228, in run
    return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/salt/loader/lazy.py", line 1243, in _run_as
    return _func_or_method(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/salt/executors/direct_call.py", line 10, in execute
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/salt/loader/lazy.py", line 149, in __call__
    return self.loader.run(run_func, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/salt/loader/lazy.py", line 1228, in run
    return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/salt/loader/lazy.py", line 1243, in _run_as
    return _func_or_method(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/salt/utils/napalm.py", line 512, in func_wrapper
    ret = func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/salt/modules/napalm_mod.py", line 980, in junos_cli
    command, format=format, dev_timeout=dev_timeout, dest=dest, **kwargs
  File "/usr/local/lib/python3.7/site-packages/salt/loader/lazy.py", line 149, in __call__
    return self.loader.run(run_func, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/salt/loader/lazy.py", line 1228, in run
    return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/salt/loader/lazy.py", line 1243, in _run_as
    return _func_or_method(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/salt/modules/junos.py", line 156, in wrapper
    ldev_timeout = max(kwargs.pop("dev_timeout", 0), kwargs.pop("timeout", 0))
TypeError: '>' not supported between instances of 'int' and 'NoneType'
```

**Expected behavior**
Expecting the junos execution modul not to throw a TypeError when looking at dev_timeout.

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.)

```yaml
Salt Version:
          Salt: 3005.1

Dependency Versions:
          cffi: 1.14.6
      cherrypy: unknown
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.1.0
       libgit2: 1.1.0
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: 2.17
      pycrypto: Not Installed
  pycryptodome: 3.9.8
        pygit2: 1.6.1
        Python: 3.7.13 (default, Apr 20 2022, 03:16:50)
  python-gnupg: 0.4.8
        PyYAML: 5.4.1
         PyZMQ: 18.0.1
         smmap: Not Installed
       timelib: 0.2.4
       Tornado: 4.5.3
           ZMQ: 4.3.1

System Versions:
          dist: alpine 3.14.6
        locale: UTF-8
       machine: x86_64
       release: 4.18.0-372.9.1.el8.x86_64
        system: Linux
       version: Alpine Linux 3.14.6

```
</details>

### PR DESCRIPTION
**Pull request by codekuu**
Removed dev_timeout from junos_cli which passes None as a variable. max() does not like None variable which makes this input mandatory when using junos_cli in salt.

Did also merge the __timeout_decorator and __timeout_decorator_cleanargs into one with comments and more readable code.
This sovles the issue with problems running junos_cli and other commands where the __pubs are exposed and max() is breaking the function since dev_timeout is provided all the time (as None) even if not provided in cli.

### What does this PR do?
Fix issues related to napalm junos_cli and other methods.

### What issues does this PR fix or reference?
Fixes: #63599 #58108

### Previous Behavior
junos_cli was breaking due to timeout_decorator was provided with value None from the dev_timeout variable used in max()

### New Behavior
Working

### Merge requirements satisfied?
- [N/A] Docs
- [N/A] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [N/A] Tests written/updated

### Commits signed with GPG?
No

### CODE REVIEW

**Comment by codekuu** [comment]
Tests and documentation is already correct for these fixes therefor no new tests or docs.

**Review by twangboy — DISMISSED** [review]

**Comment by codekuu** [comment]
Pinging this

File: salt/modules/junos.py [hunk 1]
```diff
@@ -150,75 +150,50 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
...
-            if restore_kwargs:
-                kwargs.update(op)
+            timeout_changed = True
 
-            return ret
+        # Run the function
+        try:
+            result = function(*args, **kwargs)
+        except:
+            raise
```
Comment (s0undt3ch) [inline salt/modules/junos.py:189]:
```suggestion
```
Don't catch exceptions just to raise them afterwards without doing anything with that information.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: salt/modules/junos.py (hunk 1 above)
Comment (codekuu) [reply salt/modules/junos.py:189]: It was done like that before therefor I keept the same functionality.

Comment (s0undt3ch) [reply salt/modules/junos.py:189]: I have to disagree.
Neither [this](https://github.com/saltstack/salt/blob/3006.x/salt/modules/junos.py#L153-L169) chunk, nor [this](https://github.com/saltstack/salt/blob/3006.x/salt/modules/junos.py#L174-L203) has an exception caught just to be raised immediately.
There's code that runs, before raising the exception again.

Please remove that bare except, you're not doing anything with it.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Comment by codekuu** [comment]
Removed the except: raise part but kept the finally

File: salt/modules/junos.py [hunk 2]
```diff
@@ -150,75 +150,48 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
...
+    """
+    Removing __pub kwargs during run and adding it back after to ensure non exposure.
+    Setting the timeout for the connection during run and then resetting the timeout.
+    Timeout has to be re-added for proxys running on the actual juniper
+    """
+
     @wraps(function)
     def wrapper(*args, **kwargs):
+        timeout_changed = False
+        old_kwargs = kwargs
```
Comment (s0undt3ch) [inline salt/modules/junos.py:163]: You actually need to create a copy, most likely, even a `copy.deepcopy` of the dictionary, otherwise changing `kwargs` will also change `old_kwargs`

**Review by s0undt3ch — CHANGES_REQUESTED** [review]
We also need test coverage, the bare minimum is unit testing the decorator code path.
